### PR TITLE
Improved AWS Batch guide

### PR DESCRIPTION
**Pull request by manuelesimi**
Hi,

I recently started to run my pipelines on AWS Batch for the first time. 

I followed the Nextflow's AWS cloud guide and I found out that there were a few missing information, others were outdated and others scattered across the guide. Based on this experience, I tried to reorganize the guide, creating the flow I'd have expected and adding what I believe would help other users to jump start with AWS batch. I also removed a few typos here and there.

Wherever it was possible, I aligned the structure to the Azure Batch guide.

Thanks,
manuele

### CODE REVIEW

File: docs/awscloud.rst [hunk 1]
```diff
@@ -120,26 +167,25 @@ If you don't use Conda and install the AWS CLI using something like `pip <https:
...
     wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
     bash Miniconda3-latest-Linux-x86_64.sh -b -f -p $HOME/miniconda
     $HOME/miniconda/bin/conda install -c conda-forge -y awscli
     rm Miniconda3-latest-Linux-x86_64.sh
 
-When complete verifies that the AWS CLI package works correctly::
+When complete, verify that the AWS CLI package works correctly::
 
     $ ./miniconda/bin/aws --version
-    aws-cli/1.11.120 Python/3.6.3 Linux/4.9.43-17.39.amzn1.x86_64 botocore/1.5.83
```
Comment (abhi18av) [inline docs/awscloud.rst:134]: I think that it might be better if we move to `v2` of the `aws` CLI?

Comment (manuelesimi) [reply docs/awscloud.rst:134]: @abhi18av That's a good point.

I tried to install awscliv2, but there are 2 consequences:

1) this version of the CLI seems to run inside a docker container:
<pre>
sh-4.2$ ./miniconda/bin/awsv2 --version
1.5.0
AWS CLI v2 command: docker run -i --rm -v /home/ec2-user/.aws:/root/.aws -v /home/ec2-user:/aws amazon/aws-cli
aws-cli/2.2.7 Python/3.8.8 Linux/4.14.231-173.361.amzn2.x86_64 docker/x86_64.amzn.2 prompt/off
</pre>
Would that work when called within the task's container?

2) the command name has changed, it's not `aws`, but `awsv2`. So, a switch to this new version (provided it is backward compatible) would require changes in nextflow or suggesting the user to create a symlink named `aws` to `awsv2`.

More in general, I believe that deciding whether to move to CLI2 (and probably also to ASW SDK 2, quite overdue IMHO) goes beyond the scope of this PR. Here, I was merely focusing on the documentation.

Comment (pditommaso) [reply docs/awscloud.rst:134]: thanks manule, the aws cli does not work on alpine (without some hacks), fair to stay with miniconda installation I would say.

Comment (manuelesimi) [reply docs/awscloud.rst:134]: > thanks manule, the aws cli does not work on alpine (without some hacks), fair to stay with miniconda installation I would say.

I agree.

Comment (manuelesimi) [reply docs/awscloud.rst:134]: > thanks manule, the aws cli does not work on alpine (without some hacks), fair to stay with miniconda installation I would say.

btw, I forgot to mention that my installation of cli 2 was also done with miniconda. I knew about the alpine issues.

Comment (pditommaso) [reply docs/awscloud.rst:134]: I would say to stay on v1 then

Comment (manuelesimi) [reply docs/awscloud.rst:134]: > I would say to stay on v1 then

Agree, again :)

**Review by pditommaso — APPROVED** [review]
Excellent! 👏

**Pull request merged**